This patch make the stpio_configure_pin() function available to drivers
who need to modify the behaviour of an already allocated pin.

This resolves https://bugzilla.stlinux.com/show_bug.cgi?id=668

Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
Index: linux/drivers/char/stpio.c
===================================================================
--- linux.orig/drivers/char/stpio.c
+++ linux/drivers/char/stpio.c
@@ -140,7 +140,7 @@ static spinlock_t stpio_lock = SPIN_LOCK
 		pinno = offset & 7;			\
 	} while (0)
 
-static void stpio_configure_pin(struct stpio_pin* pin, int direction)
+void stpio_configure_pin(struct stpio_pin* pin, int direction)
 {
 	STPIO_PIN_DETAILS(pin, port, pinno);
 
@@ -414,6 +414,7 @@ static void __exit stpio_exit(void)
 module_init(stpio_init);
 module_exit(stpio_exit);
 
+EXPORT_SYMBOL(stpio_configure_pin);
 EXPORT_SYMBOL(stpio_request_pin);
 EXPORT_SYMBOL(stpio_request_set_pin);
 EXPORT_SYMBOL(stpio_free_pin);
Index: linux/include/linux/stpio.h
===================================================================
--- linux.orig/include/linux/stpio.h
+++ linux/include/linux/stpio.h
@@ -50,6 +50,9 @@ void stpio_free_pin(struct stpio_pin* pi
 void stpio_set_pin(struct stpio_pin* pin, unsigned int value);
 unsigned int stpio_get_pin(struct stpio_pin* pin);
 
+/* Change the mode of an existing pin */
+void stpio_configure_pin(struct stpio_pin* pin, int direction);
+
 /* Interrupt on external value change */
 void stpio_request_irq(struct stpio_pin* pin, int mode,
 		       void (*handler)(struct stpio_pin *pin, void *dev),
